include/circt/Dialect/RTL/Combinatorial.td

```diff
@@ -79,7 +79,15 @@ class UTVariadicRTLOp<string mnemonic, list<OpTrait> traits = []> :
                !listconcat(traits,
                            [SameTypeOperands, SameOperandsAndResultType])> {
 
+  let hasCanonicalizer = 1;
   let hasFolder = 1;
+  let verifier = [{
```

Comment:
Nice, thank you for this.

Please pull it out to an out-of-line method (e.g. see how the other custom verifier hooks are defined in the FIRRTL dialect), and possible a separate patch

Reply:
Please also add a failing testcase for this to test/rtl/errors.mlir

Reply:
added negative test for zero operands case